### PR TITLE
EVG-14326 don't restart unstarted versions in commit queue

### PR DESCRIPTION
**Pull request by john-m-liu**
nope, the problem is actually that users are either manually restarting things in the commit queue, or pushing to prs that abort the commit queue merge

### CODE REVIEW

File: model/task_lifecycle.go [hunk 1]
```diff
@@ -705,7 +705,14 @@ func RestartItemsAfterVersion(cq *commitqueue.CommitQueue, project, version, cal
 		}
 		if item.Version == version {
 			foundItem = true
-		} else if foundItem {
+		} else if foundItem && item.Version != "" {
```
Comment (ablack12) [inline model/task_lifecycle.go:708]: I don't think this makes the change you want it to. If item.Version is empty, then it returns at 704, so we should only get here if item.Version != ""

File: model/patch_lifecycle.go [hunk 2]
```diff
@@ -537,7 +537,7 @@ func AbortPatchesWithGithubPatchData(createdBefore time.Time, closed bool, newPa
 
 	catcher := grip.NewSimpleCatcher()
 	for i, _ := range patches {
-		if patches[i].Version != "" {
+		if patches[i].Version != "" && patches[i].Alias != evergreen.CommitQueueAlias {
```
Comment (john-m-liu) [inline model/patch_lifecycle.go:540]: you could make the argument that if someone pushes to a pr that has a running cq merge, the active merge should abort. Thinking about this more I'm actually torn if we should either leave the merge as is, or dequeue it for the user

File: model/task_lifecycle.go (hunk 1 above)
Comment (john-m-liu) [reply model/task_lifecycle.go:708]: good catch! I left the check here as a sanity check

File: units/commit_queue.go [hunk 3]
```diff
@@ -281,14 +281,18 @@ func (j *commitQueueJob) TryUnstick(ctx context.Context, cq *commitqueue.CommitQ
...
-				"job_id":  j.ID(),
+				"message":  "merge task is not dispatchable",
+				"project":  mergeTask.Project,
+				"task":     mergeTask.Id,
+				"active":   mergeTask.Activated,
+				"priority": mergeTask.Priority,
+				"source":   "commit queue",
+				"job_id":   j.ID(),
 			})
+			j.dequeue(cq, nextItem)
```
Comment (john-m-liu) [inline units/commit_queue.go:294]: this will automatically fix the spam of alerts we got last night

File: model/patch_lifecycle.go (hunk 2 above)
Comment (ablack12) [reply model/patch_lifecycle.go:540]: I think it makes sense to dequeue it for the user; one day someone is probably going to be annoyed that it didn't

**Review by ablack12 — APPROVED** [review]

**Review by ablack12 — APPROVED** [review]

**Pull request merged**